Framework/API/inc/MantidAPI/WorkspaceListProperty.h

```diff
@@ -0,0 +1,184 @@
+  * @param workspaces: Workspaces to hold
+  * @param direction : Property directoin
+  * @param optional : Optional or mandatory property
+  * @param validator : Validator to use
+  */
+  explicit WorkspaceListProperty(
+      const std::string &name, const WorkspaceListPropertyType workspaces,
+      const unsigned int direction = Mantid::Kernel::Direction::Input,
+      const PropertyMode::Type optional = PropertyMode::Mandatory,
+      Mantid::Kernel::IValidator_sptr validator =
+          Mantid::Kernel::IValidator_sptr(new Kernel::NullValidator))
+      : Mantid::Kernel::PropertyWithValue<WorkspaceListPropertyType>(
+            name, WorkspaceListPropertyType(0), validator, direction),
+        m_optional(optional) {
+    SuperClass::operator=(workspaces);
+    auto errorMsg = isValid();
+    if (!errorMsg.empty())
+      throw std::invalid_argument(errorMsg);
+  }
+
+  /// Copy constructor, the default name stored in the new object is the same as
+  /// the default name from the original object
+  WorkspaceListProperty(const WorkspaceListProperty &right)
+      : SuperClass(right), m_optional(right.m_optional) {}
+
+  /**
+  * Assignment overload
+  * @param right : rhs workspace list property.
+  */
+  WorkspaceListProperty &operator=(const WorkspaceListProperty &right) {
+    if (&right != this) {
+      SuperClass::operator=(right);
+      m_optional = right.m_optional;
+    }
+    return *this;
+  }
+
+  /**
+  * Assignment overload
+  * @param right : rhs workspace list property type.
+  */
+  WorkspaceListPropertyType &
+  operator=(const WorkspaceListPropertyType &right) override {
+    return SuperClass::operator=(right);
+  }
+
+  /**
+  * Clone operation.
+  */
+  WorkspaceListProperty<TYPE> *clone() const override {
+    return new WorkspaceListProperty<TYPE>(*this);
+  }
+
+  // const WorkspaceListPropertyType  &operator() const { return
+  // SuperClass::operator()(); }
+  const WorkspaceListPropertyType &list() const {
+    return SuperClass::operator()();
+  }
+
+  /** Get the name of the workspace
+  *  @return The workspace's name
+  */
+  std::string value() const override { return ""; }
+
+  /** Get the value the property was initialised with -its default value
+  *  @return The default value
+  */
+  std::string getDefault() const override { return ""; }
+
+  /// Is the input workspace property optional?
+  bool isOptional() const { return m_optional == PropertyMode::Optional; }
+
+  std::string isValid() const override {
+    std::string error;
+    auto workspaces = SuperClass::m_value;
+
+    for (auto &wksp : workspaces) {
+      auto group = boost::dynamic_pointer_cast<WorkspaceGroup>(wksp);
+
+      if (group) {
+        auto names = group->getNames();
+        if (std::any_of(names.cbegin(), names.cend(), [](const std::string &s) {
+              return AnalysisDataService::Instance().doesExist(s);
+            })) {
+          error = "WorkspaceGroups which exist in the ADS are not allowed.";
```

Comment:
Good. Can you put a code comment around this to explain why we are doing it. Someone reading this might be wondering why you would enforce such a restriction.